src/make_watertight/Arc.cpp

```diff
@@ -0,0 +1,713 @@
+        delete[] my_edges;
+        MB_CHK_SET_ERR(moab::MB_FAILURE, "cannot delete edge");
+      }
+      if (debug) {
+        std::cout << "remove_opposite_edges: deleting " << duplicate_edges.size()
+                  << " edges" << std::endl;
+      }
+      duplicate_edges.clear();
+    }
+    j = i;
+  }
+
+  delete[] my_edges;
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Arc::get_next_oriented_edge(const moab::Range edges,
+                                            const moab::EntityHandle edge,
+                                            moab::EntityHandle& next_edge) {
+
+  // get the back vertex
+  moab::ErrorCode result;
+  const moab::EntityHandle* end_verts;
+  int n_verts;
+  result = MBI()->get_connectivity(edge, end_verts, n_verts);
+  assert(moab::MB_SUCCESS == result);
+  assert(2 == n_verts);
+
+  // get the edges adjacent to the back vertex
+  moab::Range adj_edges;
+  result = MBI()->get_adjacencies(&(end_verts[1]), 1, 1, false, adj_edges);
+  assert(moab::MB_SUCCESS == result);
+
+  // keep the edges that are part of the input range
+  adj_edges = intersect(adj_edges, edges);
+  // don't want the input edge
+  adj_edges.erase(edge);
+
+  // make sure the edge is oriented correctly
+  for (moab::Range::iterator i = adj_edges.begin(); i != adj_edges.end(); i++) {
+    const moab::EntityHandle* adj_end_verts;
+    result = MBI()->get_connectivity(*i, adj_end_verts, n_verts);
+    if (moab::MB_SUCCESS != result) {
+      MBI()->list_entity(*i);
+      std::cout << "result=" << result
+                << " could not get connectivity of edge" << std::endl;
+      return result;
+      //print_edge( *i );
+    }
+    assert(moab::MB_SUCCESS == result);
+    assert(2 == n_verts);
+    if (end_verts[1] != adj_end_verts[0])
+      i = adj_edges.erase(i) - 1;
+  }
+
+  /* The next edge could be ambiguous if more than one exists.This happens in
+     surfaces that are ~1D, and in surfaces that have pinch points
+     (mod13surf881). Although I didn't handle this case yet, if it occurs:
+     -Remember that a pinch point could have not just 2, but multiple ears.
+     -Select the next edge as the edge that shares the same triangle.
+     -This approach will only work if the pinch point also exists in the
+      geometric curves.
+     -If the pinch point does not exist in the geometric curves (~1D surfs),
+      there is no robust way to handle it. There should be an input assumption
+      that this never happens. "The faceting skin cannot have pinch points
+      unless they also occur in the surface's geometric curves."
+   */
+  if (0 == adj_edges.size()) {
+    next_edge = 0;
+  } else if (1 == adj_edges.size()) {
+    next_edge = adj_edges.front();
+  } else {
+    std::cout << "get_next_oriented_edge: " << adj_edges.size() <<
+              " possible edges indicates a pinch point." << std::endl;
+    result = MBI()->list_entity(end_verts[1]);
+    //assert(moab::MB_SUCCESS == result);
```

Comment:
Readability: Remove commented out code.  For the case of `assert` calls you want to keep, leave a note under what condition they should be used.